### PR TITLE
DM-17211 TAP Search Layout

### PR DESCRIPTION
**Pull request by tgoldina**
This is part 1 of DM-17211, which sets the layout and allows other people to work on the panel. 

https://jira.lsstcorp.org/browse/DM-17211

http://localhost:8080/firefly/firefly-dev.html?__action=layout.showDropDown&visible=true&view=TestTAPSearch

### CODE REVIEW

**Comment by robyww** [comment]
>  Why is it under visualize.ui? Since there will be more than a few files, I would suggest putting them under ui.tap.

I think `ui.tap` is a great idea.

File: src/firefly/js/visualize/ui/TableSelectViewPanel.jsx [hunk 1]
```diff
@@ -49,9 +54,34 @@ export class TapSearchPanel extends PureComponent {
...
+        const placeholder = serviceUrl ? `Using <${serviceUrl}>. Replace...` : 'Select TAP...';
+
         return (
             <div style={{padding: 10}}>
-                <ListBoxInputFieldView
+                <div style={{paddingBottom:5}}>
+                    <CreatableSelect
+                        options={TAP_SERVICE_OPTIONS}
+                        isClearable={true}
+                        onChange={(selectedOption) => {
```
Comment (robyww) [inline src/firefly/js/visualize/ui/TableSelectViewPanel.jsx]: I think long function inside the JSX make it hard to read.  If you need a closure function you declare it above or just make a static function that you use a closure to pass all the parameters to.

File: src/firefly/js/visualize/ui/TableSelectViewPanel.jsx [hunk 2]
```diff
@@ -65,35 +95,48 @@ export class TapSearchPanel extends PureComponent {
...
-                            </button>
-                        </div>
-                        <TapSchemaBrowser serviceUrl={serviceUrl}/>
+                        <FieldGroupTabs initialState={{ value:'ui' }} fieldKey='tabs' resizable={true}>
+                            <Tab name='Set Parameters' id='ui'>
+                                <div style={tabWrapper}>
+                                    <TapSchemaBrowser serviceUrl={serviceUrl}/>
+                                    <div style={{position: 'absolute', top: 20, right: 20}}>
+                                        <button style={{padding: 3}}
+                                                onClick={ () => {
```
Comment (robyww) [inline src/firefly/js/visualize/ui/TableSelectViewPanel.jsx]: same here.

**Comment by cwang2016** [comment]
Review completes. It sets the good foundation for filing in the spatial/temporal search part.

**Comment by tgoldina** [comment]
In the latest commit: 
- Moved TAP related panels to ui.tap
- Moved "Populate and edit ADQL" under the form
- Separated callback implementation from jsx

**Pull request merged**